pkg/dns/dns.go

```diff
@@ -323,6 +320,76 @@ func (kd *KubeDNS) handleEndpointAdd(obj interface{}) {
 	}
 }
 
+func (kd *KubeDNS) handleEndpointUpdate(oldObj, newObj interface{}) {
+	if o, ok := oldObj.(*v1.Endpoints); ok {
```

Comment:
short circuit these and log error and return if the type cast does not work. For longer scoped variables, I prefer more than one letter names.

```go
func (kd *KubeDNS) handleEndpointUpdate(oldObj, newObj interface{}) {
  oldEndpoints, ok ;= oldObj.(*v1...)
  newEndpoints, ok := newObj.(*v1...)
  if oldEndpoints == nil { 
    glog.Errorf("oldObj was not of type v1.Endpoints)
    // etc
  }
// etc